hardware_interface/src/resource_manager.cpp

```diff
@@ -1193,7 +1389,26 @@ bool ResourceManager::perform_command_mode_switch(
     return true;
   }
 
-  auto call_perform_mode_switch = [&start_interfaces, &stop_interfaces](auto & components)
+  auto interfaces_to_string = [&]()
+  {
+    std::stringstream ss;
+    ss << "Start interfaces: " << std::endl << "[" << std::endl;
+    for (const auto & start_if : start_interfaces)
+    {
+      ss << "  " << start_if << std::endl;
+    }
+    ss << "]" << std::endl;
+    ss << "Stop interfaces: " << std::endl << "[" << std::endl;
+    for (const auto & stop_if : stop_interfaces)
+    {
+      ss << "  " << stop_if << std::endl;
+    }
+    ss << "]" << std::endl;
+    return ss.str();
+  };
```

Comment:
I'd prefer if this could be a named function in an anonymous namespace above instead of a duplicated lambda

Reply:
Sure! I'll get it done tonight!

Reply:
Done! I've just pushed the changes!